src/main/scala/com.snowplowanalytics/weather/providers/openweather/Errors.scala

```diff
@@ -19,39 +19,43 @@ object Errors {
   /**
    * Superclass for non-fatal exceptions that can be happen for weather fetching/processing
    */
-  sealed trait WeatherError {
-    val message: String
-    override def toString = s"OpenWeatherMap ${this.getClass.getSimpleName} " + message
+  sealed abstract class WeatherError(message: String) extends Throwable(message) {
+    override def toString = s"OpenWeatherMap ${this.getClass.getSimpleName}: $message"
   }
 
   /**
    * Connecting/receiving/etc timeout errors
    */
-  case class TimeoutError(message: String) extends java.util.concurrent.TimeoutException(message) with WeatherError
+  final case class TimeoutError(message: String) extends WeatherError(message)
 
   /**
    * Invalid state/argument/response
    */
-  case class InternalError(message: String) extends WeatherError
+  final case class InternalError(message: String) extends WeatherError(message)
 
   /**
    * Response parsing error
    */
-  case class ParseError(message: String) extends WeatherError
+  final case class ParseError(message: String) extends WeatherError(message)
 
   /**
    * Common Auth error
    * We could also use [[ErrorResponse]], but for some unauth cases OWM returns HTML page
    */
-  case object AuthorizationError extends WeatherError {
-    val message = "Check your API key"
-  }
+  final case object AuthorizationError extends WeatherError("Check your API key")
 
   /**
    * Error returned from weather provider, which can be extracted from JSON
    */
-  @JsonCodec case class ErrorResponse(cod: Option[String], message: String) extends WeatherError
+  @JsonCodec final case class ErrorResponse(cod: Option[String], message: String) extends WeatherError(message)
+
+  /**
+   * Error linked to http but not linked to auth
+   */
+  final case class HTTPError(message: String) extends WeatherError(message)
 
-  /** Error linked to http but not linked to auth */
-  case class HTTPError(message: String) extends WeatherError
+  /**
+   * Error thrown when any argument provided to the clients is invalid
+   */
+  final case class InvalidConfigurationError(message: String) extends WeatherError(message)
```

Comment:
do we want to reuse the same adt for recoverable and non-recoverable errors? cc @chuwy

Reply:
I'd say - yes. I don't think its worth to introduce whole new ADT (potentially consisting of two-three members) for such a small domain-library. User can decide himself what is recoverable and what is not in his app.